free/src/test/scala/cats/free/CofreeTests.scala

```diff
@@ -0,0 +1,118 @@
+package cats
+package free
+
+import cats.data.NonEmptyList
+import cats.tests.CatsSuite
+import cats.laws.discipline.{CartesianTests, ComonadTests, FoldableTests, SerializableTests, TraverseTests}
+import cats.syntax.list._
+import org.scalacheck.rng.Seed
+import org.scalacheck.{Arbitrary, Cogen, Gen}
+
+class CofreeTests extends CatsSuite {
+
+  import CofreeTests._
+
+  implicit val iso1 = CartesianTests.Isomorphisms.invariant[Cofree[Option, ?]]
+  implicit val iso2 = CartesianTests.Isomorphisms.invariant[Cofree[List, ?]]
+
+  checkAll("Cofree[Option, ?]", ComonadTests[Cofree[Option, ?]].comonad[Int, Int, Int])
+  checkAll("Cofree[Option, ?]", FoldableTests[Cofree[Option, ?]].foldable[Int, Int])
+  checkAll("Cofree[Option, ?]", TraverseTests[Cofree[Option, ?]].traverse[Int, Int, Int, Int, Option, Option])
+  checkAll("Comonad[Cofree[Option, ?]]", SerializableTests.serializable(Comonad[Cofree[Option, ?]]))
+  checkAll("Cofree[List, ?]", ComonadTests[Cofree[List, ?]].comonad[Int, Int, Int])
+  checkAll("Cofree[List, ?]", FoldableTests[Cofree[List, ?]].foldable[Int, Int])
+  checkAll("Cofree[List, ?]", TraverseTests[Cofree[List, ?]].traverse[Int, Int, Int, Int, Option, Option])
```

Comment:
Do we need to test these given the tests above? Seems the only difference is s/Option/List